dcos_installer/action_lib.py

```diff
@@ -371,38 +371,129 @@ def uninstall_dcos(config, block=False, state_json_dir=None, async_delegate=None
-  echo "install_prereqs has been already executed on this host, exiting..."
-  exit 0
+echo "Validating kernel version..."
+kernel_major_version="$(uname -r | sed -e 's/\([0-9][0-9]*\).*/\1/')"
+kernel_minor_version="$(uname -r | sed -e "s/${kernel_major_version}\.\([0-9][0-9]*\).*/\1/")"
+echo "Kernel Version: ${kernel_major_version}.${kernel_minor_version}"
+if [[ ${kernel_major_version} -lt 1 || ${kernel_major_version} -eq 1 && ${kernel_minor_version} -lt 11 ]]; then
+  echo "Error: Kernel version ${kernel_major_version}.${kernel_minor_version} is not supported. "\
+"Only >= 3.10 is supported." >&2
+  exit 1
 fi
 
-sudo setenforce 0 && \
+echo "Validating kernel modules..."
+if ! lsmod | grep -q overlay; then
+  echo "Enabling OverlayFS kernel module..."
+  # Enable now
+  sudo modprobe overlay
+  # Load on reboot via systemd
+  sudo tee /etc/modules-load.d/overlay.conf <<-'EOF'
+overlay
+EOF
+fi
+
+echo "Validating file system..."
+sudo mkdir -p /var/lib/docker
+file_system="$(df --output=fstype /var/lib/docker | tail -1)"
+echo "File System: ${file_system}"
+if [[ "${file_system}" != 'xfs' ]] || ! xfs_info /var/lib/docker | grep -q 'ftype=1'; then
+  echo "Error: /var/lib/docker must use XFS provisioned with ftype=1 to avoid known issues with OverlayFS." >&2
+  exit 1
+fi
+
+echo "Installing Utilities..."
+sudo yum install -y wget
+sudo yum install -y curl
+sudo yum install -y git
+sudo yum install -y unzip
+sudo yum install -y xz
+sudo yum install -y ipset
+
+echo "Disabling SELinux..."
+sudo setenforce 0
 sudo sed -i --follow-symlinks 's/^SELINUX=.*/SELINUX=disabled/g' /etc/sysconfig/selinux
 
-sudo tee /etc/yum.repos.d/docker.repo <<-'EOF'
+echo "Detecting Docker..."
+install_docker='true'
+if hash docker 2>/dev/null; then
+  docker_client_version="$(docker --version | sed -e 's/Docker version \(.*\),.*/\1/')"
+  echo "Docker Client Version: ${docker_client_version}"
+
+  if ! docker info &>/dev/null; then
+    echo "Docker Server not found. Please uninstall Docker and try again." >&2
+    exit 1
+  fi
+
+  docker_server_version="$(docker info | grep 'Server Version:' | sed -e 's/Server Version: \(.*\)/\1/')"
+  echo "Docker Server Version: ${docker_server_version}"
+
+  if [[ "${docker_client_version}" != "${docker_server_version}" ]]; then
+    echo "Docker Server and Client versions do not match. Please uninstall Docker and try again." >&2
+    exit 1
+  fi
+
+  # Require Docker >= 1.11
+  docker_major_version="$(echo "${docker_server_version}" | sed -e 's/\([0-9][0-9]*\)\.\([0-9][0-9]*\).*/\1/')"
+  docker_minor_version="$(echo "${docker_server_version}" | sed -e 's/\([0-9][0-9]*\)\.\([0-9][0-9]*\).*/\2/')"
+  if [[ ${docker_major_version} -lt 1 || ${kernel_minor_version} -lt 11 ]]; then
+    echo "Docker version ${docker_server_version} not supported. Please uninstall Docker and try again." >&2
+    exit 1
+  fi
+
+  install_docker='false'
+fi
+
+if [[ "${install_docker}" == 'true' ]]; then
+  echo "Installing Docker..."
```

Comment:
It might be helpful if this output included the version of Docker we're installing.

Reply:
I had that originally but then there was more than one place to change the version... the yum install output should say which version is installed